pkg/controllers/framework/factory.go

```diff
@@ -18,19 +18,49 @@ package framework
 
 import (
 	"fmt"
+	"strings"
 
 	"k8s.io/klog/v2"
 )
 
 var controllers = map[string]Controller{}
 
 // ForeachController is helper function to operator all controllers.
-func ForeachController(fn func(controller Controller)) {
+func ForeachController(controllerGate string, fn func(controller Controller)) {
 	for _, ctrl := range controllers {
-		fn(ctrl)
+		// if controllerGate is "*", all controllers will be executed.
+		if controllerGate == "*" || checkControllerGate(controllerGate, ctrl.Name()) {
+			fn(ctrl)
+		}
 	}
 }
 
+func checkControllerGate(controllerGate, controllerName string) bool {
```

Comment:
A little complicated here, you can refer the mentioned link of the issue: )

Reply:
done